src/main/java/uk/gov/pay/ledger/transaction/model/PaymentFactory.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.pay.ledger.transaction.model;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.ledger.transaction.entity.TransactionEntity;
+import uk.gov.pay.ledger.transaction.search.model.RefundSummary;
+import uk.gov.pay.ledger.transaction.search.model.SettlementSummary;
+import uk.gov.pay.ledger.transaction.state.TransactionState;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+public class PaymentFactory {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(PaymentFactory.class);
+    private ObjectMapper objectMapper;
+
+    @Inject
+    public PaymentFactory(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public Payment fromTransactionEntity(TransactionEntity entity) {
+        try {
+            JsonNode transactionDetails = objectMapper.readTree(Optional.ofNullable(entity.getTransactionDetails()).orElse("{}"));
+            Address billingAddress = Address.from(
+                    safeGetAsString(transactionDetails, "address_line1"),
+                    safeGetAsString(transactionDetails, "address_line2"),
+                    safeGetAsString(transactionDetails, "address_postcode"),
+                    safeGetAsString(transactionDetails, "address_city"),
+                    safeGetAsString(transactionDetails, "address_county"),
+                    safeGetAsString(transactionDetails, "address_country")
+            );
+
+            CardDetails cardDetails = CardDetails.from(entity.getCardholderName(), billingAddress, entity.getCardBrand(),
```

Comment:
Again not sure about `from` pattern, I might be wrong on this!

Reply:
Again it can't.
https://github.com/alphagov/pay-ledger/blob/master/src/main/java/uk/gov/pay/ledger/transaction/model/CardDetails.java#L30-L42

Reply:
Understood! Maybe having this as optional throughout the code would make this clear